Xamarin.Forms.DataGrid/DataGrid.xaml.cs

```diff
@@ -417,6 +423,27 @@ public DataGrid(ListViewCachingStrategy cachingStrategy)
 		}
 		#endregion
 
+		//2018.03.29, gwise@naver.com, Double Tap Function Add.
+		private int selectedCount = 0;
+		private ItemTappedEventArgs itemTappedEventArgs;
+		private void DataGrid_ItemTapped(object sender, ItemTappedEventArgs e)
```

Comment:
Logic seems complicated and needs to be simplified. You might use TapGestureRecognizer.